spec/integration/api_resource_comments_controller_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'swagger_helper'
+
+describe ApiResourceCommentsController do
+  let(:user) { create(:user) }
+  let(:resource) { create(:resource, user:) }
+  let(:resource_id) { resource.id }
+
+  before do
+    sign_in user
+  end
+
+  path '/api/resources/{resource_id}/resource_comments' do
+    get 'Retrieves a list of resource comments' do
+      tags 'resource_comments'
+      operationId 'getResourceComments'
+      produces 'application/json'
+      parameter name: :resource_id, in: :path, type: :string
+
+      response '200', 'Success' do
+        schema type: :array,
+               items: {
+                 properties: {
+                   id: { type: :integer },
+                   user_id: { type: :integer },
+                   resource_id: { type: :integer },
+                   content: { type: :string }
+                 }
+               }
+        run_test!
+      end
+
+      response '404', 'Resource Not Found' do
+        schema type: :object, properties: {
+          'code': { type: :integer },
+          'message': { type: :string },
+          'status': { type: :string }
+        }
+        let(:resource_id) { 'invalid' }
+        run_test!
+      end
+    end
+  end
+end
```

Comment:
I think it is also necessary to test the case of a resource with no comments. It's very particular, because it retrieves an empty array. Is that ok? Is that what we expect? (I think it is, but I'd test it anyway).
Maybe you could also split the cases where there is only 1 comment and when there are more than 1, but I think it's not strictly necessary.

Reply:
Agree, we should test that resources with no comments return an empty array and if at least one comment exist the array is not empty.